lib/rules/no-restricted-exports.js

```diff
@@ -41,6 +41,26 @@ module.exports = {
                     },
                     additionalProperties: false
                 },
+                {
+                    type: "object",
+                    properties: {
+                        restrictedNamedExports: {
+                            type: "array",
+                            items: {
+                                type: "string"
+                            },
+                            uniqueItems: true
+                        },
+                        restrictedNamedExportsPattern: {
+                            type: "array",
+                            items: {
+                                type: "string"
+                            },
+                            uniqueItems: true
+                        }
+                    },
+                    additionalProperties: false
+                },
```

Comment:
There's no need for another element in `anyOf`, we can add the new property `restrictedNamedExportsPattern` directly into the two existing elements.

Also, I think we agreed in https://github.com/eslint/eslint/issues/18360 that `restrictedNamedExportsPattern` will be a string (one pattern), not an array of strings.